cxx/distributions/string_nat_test.cc

```diff
@@ -0,0 +1,24 @@
+// Apache License, Version 2.0, refer to LICENSE.txt
+
+#define BOOST_TEST_MODULE test StringNat
+
+#include "distributions/string_nat.hh"
+
+#include <boost/test/included/unit_test.hpp>
+
+BOOST_AUTO_TEST_CASE(test_simple) {
+  StringNat sn;
+
+  sn.incorporate("42");
+  sn.incorporate("0");
+  BOOST_TEST(sn.N == 2);
+  sn.unincorporate("0");
+  BOOST_TEST(sn.N == 1);
+}
+
+BOOST_AUTO_TEST_CASE(test_nearest) {
```

Comment:
Is it intended the empty string is valid for this distribution (asking since a non-numeric string will return empty with nearest).

Reply:
Yes.  All this distribution is is strings formed out of digits, and those strings can totally be empty.  In retrospect, "DecimalString" might have been a better name for it than StringNat, because is meant to model things like zip codes where preserving leading 0's matters.